src/Cooked/MockChain/GenerateTx/Witness.hs

```diff
@@ -0,0 +1,87 @@
+module Cooked.MockChain.GenerateTx.Witness
+  ( toRewardAccount,
+    toScriptWitness,
+  )
+where
+
+import Cardano.Api.Shelley qualified as Cardano hiding (Testnet)
+import Cardano.Ledger.Address qualified as Cardano
+import Cardano.Ledger.BaseTypes qualified as Cardano
+import Cardano.Ledger.Credential qualified as Cardano
+import Cardano.Ledger.Crypto qualified as Crypto
+import Control.Monad
+import Control.Monad.Reader
+import Cooked.Conversion
+import Cooked.MockChain.GenerateTx.Common
+import Cooked.Output
+import Cooked.Skeleton
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Ledger.Tx.CardanoAPI qualified as Ledger
+import Optics.Core
+import Plutus.Script.Utils.Scripts qualified as Script
+import PlutusLedgerApi.V3 qualified as Api
+
+type WitnessGen a = TxGen (Map Api.TxOutRef Api.TxOut) a
+
+-- | Translates a given credential to a reward account.
+toRewardAccount :: Api.Credential -> WitnessGen (Cardano.RewardAcnt Crypto.StandardCrypto)
+toRewardAccount cred =
+  Cardano.RewardAcnt Cardano.Testnet <$> case cred of
+    Api.ScriptCredential scriptHash -> do
+      Cardano.ScriptHash cHash <-
+        throwOnToCardanoError
+          "toRewardAccount: Unable to convert script hash."
+          (Ledger.toCardanoScriptHash scriptHash)
+      return $ Cardano.ScriptHashObj cHash
+    Api.PubKeyCredential pubkeyHash -> do
+      Cardano.StakeKeyHash pkHash <-
+        throwOnToCardanoError
+          "toRewardAccount: Unable to convert private key hash."
+          -- TODO: we take the pubkeyHash, maybe we should take the
+          -- stakehash if any? I am confused about the nature of the
+          -- stake address.
```

Comment:
Maybe this should be rephrased in non personal way/removed/addressed before merging

Reply:
For example `The nature of the stake address is confusing`. The `I` seems weird in a company project.